sdk/sdk-api/src/main/java/mil/emp3/api/abstracts/Feature.java

```diff
@@ -76,6 +77,11 @@ public void addFeatures(List<IFeature> features, boolean visible)
         if (features == null) {
             throw new EMP_Exception(EMP_Exception.ErrorDetail.INVALID_PARAMETER, "Parameter to Feature.addFeatures can not be null.");
         } else if (features.size() > 0) {
+            for (IFeature feature : features) {
+                if (feature instanceof MilStdSymbol && ((MilStdSymbol) feature).getSymbolCode() == null) {
+                    throw new EMP_Exception(EMP_Exception.ErrorDetail.INVALID_PARAMETER, "Can't add feature with null symbol code");
+                }
```

Comment:
It might be useful to create a method validate() on Feature class, that would just return. We can then override it in the subclass as and when necessary, e.g. MildStdSymbol can check for null Symbold code. Method could trow IllegalState exception. Then we wouldn't never have to check specifically for MilStdSymbol class. This method can also be used with MilStdSymbol where you are currently checking for null SymbolCode.